Managers/UInputMan.cpp

```diff
@@ -981,4 +1012,74 @@ namespace RTE {
 			}
 		}
 	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifdef __unix__
+	void UInputMan::HandleAllegroMouseInput() {
+		if (_xwin.display == 0)
+			return;
+
+		int nEvents = XPending(_xwin.display);
+		std::vector<XEvent> events;
+		events.reserve(nEvents + 10);
+		for (int i = 0; i < XPending(_xwin.display); ++i) {
+			XEvent e;
+			XNextEvent(_xwin.display, &e);
+			events.push_back(e);
+		}
+
+		static int mousePrevX = 0;
+		static int mousePrevY = 0;
+		mousePrevX = mouse_x;
+		mousePrevY = mouse_y;
+		int dx = 0;
```

Comment:
Technically should be `dX` and `dY`. I'd also be totally cool with more descriptive names though lol